core-bundle/src/Security/User/ContaoUserProvider.php

```diff
@@ -107,6 +107,19 @@ public function supportsClass($class): bool
         return $this->userClass === $class;
     }
 
+    /**
+     * @param User $user
+     */
+    public function upgradePassword(UserInterface $user, string $newEncodedPassword): void
+    {
+        if (!is_a($user, $this->userClass)) {
```

Comment:
This shouldn't happen because of the `supports()` call. But anyway :)

Reply:
You are right. I have just copied the code from the `refreshUser()` method, but maybe it is unnecessary there as well?

https://github.com/contao/contao/blob/1e08698e6122029c6d8eda45010e59eb54bc2fcd/core-bundle/src/Security/User/ContaoUserProvider.php#L87-L91

Reply:
Well, defensive programming - guess that's fine :)